sound/soc/sof/sof-audio.h

```diff
@@ -106,6 +109,8 @@ struct sof_ipc_pcm_ops {
 	int (*trigger)(struct snd_soc_component *component,  struct snd_pcm_substream *substream,
 		       int cmd);
 	int (*dai_link_fixup)(struct snd_soc_pcm_runtime *rtd, struct snd_pcm_hw_params *params);
+	int (*pcm_setup)(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm);
+	void (*pcm_free)(struct snd_sof_dev *sdev, struct snd_sof_pcm *spcm);
```

Comment:
This is confusing for me. All the existing callbacks are self-explanatory, they are traditional 'ALSA' callbacks, so I can picture when they are used.

pcm_setup() is called in sof_dai_load() and pcm_free() is called in sof_dai_unload(). That's far from obvious to me what setup and load really mean... It wouldn't hurt to describe what the 'setup' is supposed to do.

Reply:
there should also be a mention that these callbacks are optional? That's how they are used but it's not clear in the kernel-doc

Reply:
@ujfalusi I felt the same but it's my understanding that these helpers allocate stuff and prepare the ground for the trigger to operate, so in a way they are companion functions to the trigger and should be in the same ops.

Reply:
the ops are invoked from topology but since they deal with spcm I think they go better with pcm_ops instead of tplg_ops.